components/suggest/src/schema.rs

```diff
@@ -45,11 +45,14 @@ CREATE TABLE prefix_keywords(
     keyword_suffix TEXT NOT NULL DEFAULT '',
     confidence INTEGER NOT NULL DEFAULT 0,
     rank INTEGER NOT NULL,
-    suggestion_id INTEGER NOT NULL REFERENCES suggestions(id) ON DELETE CASCADE,
+    suggestion_id INTEGER NOT NULL REFERENCES suggestions(id),
     PRIMARY KEY (keyword_prefix, keyword_suffix, suggestion_id)
 ) WITHOUT ROWID;
 
 CREATE UNIQUE INDEX keywords_suggestion_id_rank ON keywords(suggestion_id, rank);
+CREATE INDEX prefix_keywords_suggestion_id ON prefix_keywords(suggestion_id);
```

Comment:
Thanks for adding these indexes here! I think we didn't add them earlier because the number of MDN, AMO, and Pocket suggestions was so small compared to AMP and Wikipedia, but it's wonderful to have them for symmetry!

I'm idly curious how well this would benchmark, and if it would have any impact one way or the other on the database size:

```suggestion
CREATE UNIQUE INDEX prefix_keywords_suggestion_id ON prefix_keywords(suggestion_id, confidence, rank);
```

(Pocket keywords have two separate lists, one with low-confidence and one with high-confidence prefixed keywords, and the `rank` is the index of the keyword within that list. For AMO and MDN, the `confidence` is always 0).

Reply:
I should say explicitly: this is totally non-blocking, and there's no need to do this now if you'd like to land the fix ASAP.

Reply:
I'm going to go with your index since it seems more correct and has a negligible affect on performance.  Disk size was exactly the same -- I guess the difference fit inside the remainder of the current page.  On my system the pocket benchmark finishes in 12ms and AMO is in the nano-second range, both before and after the change.